arches/app/etl_modules/import_single_csv.py

```diff
@@ -72,13 +96,57 @@ def get_node_lookup(self, graphid):
             self.node_lookup[graphid] = Node.objects.filter(graph_id=graphid)
         return self.node_lookup[graphid]
 
-    def read(self, request):
+    def cli(self, source):
+        initiated = self.start(self.request)
+
+        if initiated["success"]:
+            read = self.read(source=source)
+        else:
+            return {
+                "success": False,
+                "data": {"title": _("Error"), "message": initiated["message"]},
+            }
+
+        if read["success"]:
+            written = self.write(self.request)
+        else:
+            return {
+                "success": False,
+                "data": {"title": _("Error"), "message": read["message"]},
+            }
+
+        if written["success"]:
+            return {"success": True, "data": "done"}
+        else:
+            return {
+                "success": False,
+                "data": {"title": _("Error"), "message": written["message"]},
+            }
+
+    def read(self, request=None, source=None):
         """
         Reads added csv file and returns all the rows
-        If the loadid already exsists also returns the load_details
+        If the loadid already exists also returns the load_details
         """
 
-        content = request.FILES.get("file")
+        if request:
+            content = request.FILES.get("file")
+        else:
+            if source.split(".")[-1].lower() == "csv":
+                file_type = "text/csv"
+            elif source.split(".")[-1].lower() == "zip":
+                file_type = "application/zip"
+            file = open(source, "rb")
```

Comment:
Seems like this should be in a context statement: ```with open(source, "rb") as file:```

Reply:
It turns out that the file needs to open a little bit longer so the opened is manually closed when it is done, now.